mono/metadata/handle.h

```diff
@@ -401,19 +401,21 @@ This is why we evaluate index and value before any call to MONO_HANDLE_RAW or ot
 #define MONO_HANDLE_ARRAY_SETREF(HANDLE, IDX, VALUE) do {	\
 		int __idx = (IDX);	\
    		MonoObjectHandle __val = MONO_HANDLE_CAST (MonoObject, VALUE);		\
-		do {							\
+		{	/* FIXME scope needed by Centrinel */		\
+			/* FIXME mono_array_setref_fast is not an expression. */ \
 			MONO_HANDLE_SUPPRESS_SCOPE(1);			\
 			mono_array_setref_fast (MONO_HANDLE_RAW (MONO_HANDLE_UNSUPPRESS (HANDLE)), __idx, MONO_HANDLE_RAW (__val)); \
-		} while (0);						\
+		}							\
 	} while (0)
 
 #define MONO_HANDLE_ARRAY_SETVAL(HANDLE, TYPE, IDX, VALUE) do {		\
 		int __idx = (IDX);					\
    		TYPE __val = (VALUE);					\
-		do {							\
+		{	/* FIXME scope needed by Centrinel */		\
+			/* FIXME mono_array_set is not an expression. */ \
```

Comment:
I don't think we want mono_array_set(_fast) to be expressions.

Reply:
Function calls are imho one the main things to model after and they are expressions. The macros look like function calls. I am I admit predisposed to want almost everything to be an expression w/o thinking about it much. You can build more on them. Even if they are void expressions. What should be a statement? Even things that are statements people complain about -- for example in some languages if/then is an expression -- kinda like a big ternary, returning their last expressions.